locale/en/locale.cfg

```diff
@@ -27,31 +55,89 @@ warehouse-active-provider=Active Provider Warehouse
 warehouse-requester=Requester Warehouse
 warehouse-buffer=Buffer Warehouse
 
+warehouse-basic-small=Small Warehouse
+warehouse-passive-provider-small=Small Passive Provider Warehouse
+warehouse-storage-small=Small Storage Warehouse
+warehouse-active-provider-small=Small Active Provider Warehouse
+warehouse-requester-small=Small Requester Warehouse
+warehouse-buffer-small=Small Buffer Warehouse
+
+warehouse-basic-tiny=Tiny Warehouse
+warehouse-passive-provider-tiny=Tiny Passive Provider Warehouse
+warehouse-storage-tiny=Tiny Storage Warehouse
+warehouse-active-provider-tiny=Tiny Active Provider Warehouse
+warehouse-requester-tiny=Tiny Requester Warehouse
+warehouse-buffer-tiny=Tiny Buffer Warehouse
+
 storehouse-basic=Storehouse
 storehouse-passive-provider=Passive Provider Storehouse
 storehouse-storage=Storage Storehouse
 storehouse-active-provider=Active Provider Storehouse
 storehouse-requester=Requester Storehouse
 storehouse-buffer=Buffer Storehouse
 
+storehouse-basic-small=Small Storehouse
+storehouse-passive-provider-small=Small Passive Provider Storehouse
+storehouse-storage-small=Small Storage Storehouse
+storehouse-active-provider-small=Small Active Provider Storehouse
+storehouse-requester-small=Small Requester Storehouse
+storehouse-buffer-small=Small Buffer Storehouse
+
+storehouse-basic-tiny=Tiny Storehouse
+storehouse-passive-provider-tiny=Tiny Passive Provider Storehouse
+storehouse-storage-tiny=Tiny Storage Storehouse
+storehouse-active-provider-tiny=Tiny Active Provider Storehouse
+storehouse-requester-tiny=Tiny Requester Storehouse
+storehouse-buffer-tiny=Tiny Buffer Storehouse
+
 [item-description]
-warehouse-basic=Stores many many items
-warehouse-passive-provider=Passive Provider Warehouse
-warehouse-storage=Storage Warehouse
-warehouse-active-provider=Active Provider Warehouse
-warehouse-requester=Requester Warehouse
-warehouse-buffer=Buffer Warehouse
```

Comment:
I'm not a fan of the new item descriptions. I'm also not a fan of the old descriptions that simply duplicated the item name, to be fair… but these new ones ("Great/Good storage capacity") don't feel right.

How would you feel about keeping the original theme ("Stores many many items" for the warehouses, and "Stores some items" or "Stores many items" for the storehouses) and updating the logistic ones to match that?

For example, the `item-description` for `warehouse-passive-provider` could be: "Stores many many items, and passively provides them to the logistic network"

The "small" and "tiny" variants could have the same description as their larger counterpart, but with some reference to being smaller or compressing the items somehow (quantum technobabble or something).

Thoughts?

Reply:
I really like your suggestion of "Stores many many items, and passively provides them to the logistic network" I just didn't like how they were all just the name as the description too so it was just an idea I had to give them all at least some time of description.  Would just have to manipulate them for the different logistics versions.  Also I thought of maybe instead of "many many items" giving an actual inventory slot count that would read from whatever the player chooses (if possible since you added in the ability to change the slot count).  But I also though this may feel clunky or awkward sounding depending on how it's written.

Reply:
A few times I've played with pulling stuff into the localized strings, and I've never gotten it to work properly. Might be I just don't know how to do it properly.

If you want to take a shot at making the item descriptions better, you can certainly update the existing ones in a separate pull request without waiting for this one to get merged. Then I can help you learn to rebase this branch onto that later, so this PR only needs to add strings for the new buildings it adds.

Separating unrelated changes is a big part of making pull requests manageable. This PR will probably sit for a while, because a big change like adding two more sets of buildings is definitely going to wait for at least a version bump to 0.3.0, if not a later release. Smaller changes like tweaking the existing strings (and others I'll comment on in a moment) can be done separately from this PR, and get integrated sooner.

Reply:
So if I wanted to go about doing another PR would I want to create a separate local clone and only make the cosmetic wire changes and description changes?

Reply:
Hmm… On the one hand, it would be instructive for you to fix your `master` branch. But I really don't know where to even start with walking you through doing that using IntelliJ.

I can give you the terminal commands if you can figure out where to `cd` to in Git Bash…

Or you can delete your local repo (**not** your fork on GitHub! that would close this PR!) and clone `Warehousing/Warehousing` fresh. That will give you the `master` branch as it should be, usable as a base for new PRs. Then you'd just have to figure out adding your own fork back in as a remote. Once you do that, pulling the `tiny-warehousing` branch back down to continue working on it should be easy enough, and you could probably figure out how to force-push the "real" `master` to your `master` through the IntelliJ GUI.

Reply:
Okay so I think i got kind of a general concept of how to make the remotes work in intellij to have the 2 different sets of code.

That being said back to the item description changes and possibly pulling in a variable to the .cfg file I can't find any possible way to do that but it may be possible to it in reverse.  I spent hours trying to figure it out on my own yesterday and I think I have an idea on what is needed but no clue how to actually make it work.

My idea is to make use of the "localised_description" class with the LuaEntityPrototype.  If we could pull data from specific lines of the .cgf file and bring them into the entity.lua file based on whatever locale is in use at the time it would allow all the locales to work as is.

The first thing we would need to do would be to would be to make a function to pull the data from the .cfg file based on the current game locale in the entity.lua file.  I believe when you make a function you're creating a variable.  So lets say we pulled in the data from whichever .cfg file is active and made it the variable **locale_item_description.**  Then make use of the localised_description class and and the fact that you already have variables set for the inventory slot counts.  Say we make the item description in the .cfg file say "slot storage capacity".

inventory_size = warehouse_slots,
localised_description = **warehouse_slot** **locale_item_description**,
scale_info_icons = settings.startup["Warehousing-icon-scaling"].value,

Reply:
I remember looking into getting locale strings via Lua and came away with the impression that we can't get to them via Lua. That could be misremembering, but it's really not _that_ big of a deal to include (or not) the capacity numbers inside the item descriptions. It's not worth spending a huge amount of time trying to do it.

Would be a much better use of time to simply think of an alternative idea and implement that, then move on to the next offshoot PR :D

Reply:
I know it's not a huge deal I just wanted to figure it out because it bothers me not knowing how to do something when logically it should be possible haha.  I like puzzles.

For the cosmetic changes to the wire positioning would you have to create another separate branch that I would push to or would we just use the tiny-warehouse branch and push those small changes to it to start with and then the major changes would come from my local branch at a later time?

Reply:
You can just make a new branch in your fork, make the wire-positioning changes, and create a separate pull request from that branch.